src/CMakeLists.txt

```
@@ -50,6 +50,20 @@ celeritas_target_include_directories(BuildFlags
     "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>"
 )
 
+#----------------------------------------------------------------------------#
+# Enable test coverage
+#----------------------------------------------------------------------------#
+if(CELERITAS_CODE_COVERAGE)
+  set(GCC_COVERAGE_COMPILE_FLAGS "-fprofile-arcs;-ftest-coverage")
+  set(GCC_COVERAGE_LINK_FLAGS    "-fprofile-arcs")
+  target_compile_options(BuildFlags
+    INTERFACE "$<$<COMPILE_LANGUAGE:CXX>:${GCC_COVERAGE_COMPILE_FLAGS}>"
+  )
+  target_link_options(BuildFlags
+    INTERFACE "${GCC_COVERAGE_LINK_FLAGS}"
+  )
```

Is this GCC specific? Does it wrk for LLVM too? Shouldn't it be `CELERITAS_GCC_CODE_COVERAGE` if it's compiler specific?

Couldn't we just have:
```suggestion
if(DEFINED CELERITAS_COMPILE_FLAGS)
  target_compile_options(BuildFlags
    INTERFACE "$<$<COMPILE_LANGUAGE:CXX>:${CELERITAS_COMPILE_FLAGS}>"
  )
endif()
if(DEFINED CELERITAS_LINK_FLAGS)
  target_link_options(BuildFlags
    INTERFACE "${CELERITAS_LINK_FLAGS}"
  )
endif()
```

and add `{"CELERITAS_COMPILE_FLAGS": "-fprofile-arcs;-ftest-coverage"},
{"CELERITAS_LINK_FLAGS": "-fprofile-arcs"}` to the cmake presets for the jobs that run it?

These just seem too specific to work in all cases for all compilers (past and future).

---

> Is this GCC specific? Does it wrk for LLVM too?

I think it originated with GCC but is indeed working with Clang.

---

>  to the cmake presets for the jobs that run it?

This is a philosophical question.  Should coverage be enable only for the CI (and/or subset of preset) or should you be able to run it locally?  You seem lean for the former, I strongly recommend/want the later.